src/openforms/submissions/api/viewsets.py

```diff
@@ -139,6 +140,8 @@ def on_submission_commit():
 
         transaction.on_commit(on_submission_commit)
 
+        book_appointment_for_submission(submission)
```

Comment:
Does this *need* to run in the request response cycle? Or could we off-load this to a celery task (in the above `on_submission_commit`?

It looks like the error-handling is already done as part of this function, so even if there is not a successful appointment registration, the end-user flow will not be interrupted now.

Reply:
> Does this need to run in the request response cycle? Or could we off-load this to a celery task (in the above on_submission_commit?

Good question :) The reason I did it like this is that in the next couple of stories related to the appointments we need to pass the appointment identifier (if there is one) so that we can add information about the appointment to the confirmation email.  This means this function needs to run before `send_confirmation_email`.  

We could off-load this to a celery task but then we'll need to do the same with the confirmation email.  If we do want to off-load both of these then we can though I wonder if it's better to do as part of this pull request or add this to one of the "improvement" tickets related to this.